lib/mongoHandler.js

```diff
@@ -144,10 +144,26 @@ MongoStore.prototype.initialise = function(resourceConfig) {
   }
   self.resourceConfig = resourceConfig;
   self.relationshipAttributeNames = MongoStore._getRelationshipAttributeNames(resourceConfig.attributes);
-  mongodb.MongoClient.connect(self._config.url).then(function(db) {
+  mongodb.MongoClient.connect(self._config.url, {
+    server: {
+      reconnectTries: 999999999,
+      reconnectInterval: 5000
+    }
+  }).then(function(db) {
     self._db = db;
+    self._db.on("close", function(err) {
+      console.error("mongodb connection closed:", err.message);
+      self.ready = false;
+      self._db.collection("Nope").findOne({ _id: 0 }, { _id: 0 }, function() {
+        console.error("mongodb connection is back");
```

Comment:
This query is expected to error, HOWEVER, given the server settings above, the callback will only fire when either:
 1. The connection is restored
 2. We give up attempting to reconnect (after.... (999999999 * 5)/(3600*24)... 57,870 days).